Tensile/Source/ReferenceCPU.h

```diff
@@ -104,113 +105,223 @@ TensileStatus tensileReferenceCPU(
   std::vector<unsigned int> coordsB( numIndicesAB );
 
   bool moreIndicesC = true;
-  while (moreIndicesC) { // iterate over entire free index range
+  if (useHighPrecisionAccumulate) {
```

Comment:
Rather than duplicate the entire loop, can you move this if so it it closer to the computation (and keep the loop the same for the userHighPrecisionAccumulate = 0)